source/components/dispatcher/dswexec.c

```diff
@@ -571,7 +571,8 @@ AcpiDsExecEndOp (
          * All opcodes require operand resolution, with the only exceptions
          * being the ObjectType and SizeOf operators.
          */
-        if (!(WalkState->OpInfo->Flags & AML_NO_OPERAND_RESOLVE))
+        if (!(WalkState->OpInfo->Flags & AML_NO_OPERAND_RESOLVE) &&
+            WalkState->NumOperands > 0)
```

Comment:
By the way, what does the ACPI spec require? Is zero operands allowed? If not, an error should be shown, the firmware bug be logged.

Reply:
Well I checked which opcode is hit and how many arguments it requires:
[    5.292010] DEBUG: drivers/acpi/acpica/dswexec.c:397 opcode = 5b33; operands = 0

According to spec it is:
0x5B 0x33 TimerOp Term Object — —
Which requires no arguments (hence "—" in spec).
That is with assumption that what spec defines as argument is interpreted as operand in code...